mlflow/models/evaluation/base.py

```diff
@@ -846,87 +846,8 @@ def _is_model_deployment_endpoint_uri(model: Any) -> bool:
 def _get_model_from_deployment_endpoint_uri(
     endpoint_uri: str, params: Optional[Dict[str, Any]] = None
 ):
-    from mlflow.metrics.genai.model_utils import _call_deployments_api, _parse_model_uri
-    from mlflow.pyfunc.model import _PythonModelPyfuncWrapper
-
-    class ModelFromDeploymentEndpoint(mlflow.pyfunc.PythonModel):
```

Comment:
note: This is a separate request to move this `ModelFromDeploymentEndpoint` out of the local function scope, so consumers can import this class and check with `isinstance(xxx, ModelFromDeploymentEndpoint)`. The only reason we did the inner class was to avoid circular import so we should have no issue with extracting this out.

Reply:
can we isolate the change into a separate PR?

Reply:
kk, will file a separate PR

Reply:
Filed https://github.com/mlflow/mlflow/pull/13230